### PR TITLE
dynamic_modules: enables rustfmt.toml

### PR DESCRIPTION
**Pull request by mathetake**
Commit Message: dynamic_modules: enables rustfmt.toml
Additional Description:

This enables the root configuration for rustfmt as a follow up
on https://github.com/envoyproxy/envoy/pull/37070#discussion_r1845515368.
Only formatting is done in this commit. There's no change in its code.

Risk Level: low
Testing: n/a 
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/dynamic_modules/test_data/rust/rustfmt.toml
Comment (mathetake) [inline test/extensions/dynamic_modules/test_data/rust/rustfmt.toml:1]: note: this is to allow editors to auto format

File: source/extensions/dynamic_modules/sdk/rust/rustfmt.toml
Comment (mathetake) [inline source/extensions/dynamic_modules/sdk/rust/rustfmt.toml:1]: same

**Comment by mathetake** [comment]
cc @phlax

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**